### PR TITLE
75174 add MAP STS logging

### PR DESCRIPTION
**Pull request by bramleyjl**
## Summary

- Adds additional logging to the MAP STS token request, including
  - a `token request` log that runs before every request for a MAP STS token
  - updates `token success` log to occur after MAP STS response is parsed
  - error logs for MAP STS-returned errors, missing ICN, and application mismatch

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/75174

## Testing done

- [x] *New code is covered by unit tests*

To test, create a [Service Account token](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/identity/Products/Sign-In%20Service/Engineering%20Docs/auth_flows/service_account.md), then use that to query the [MAP STS token endpoint](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/identity/Products/Service%20Account%20Endpoints/MAP%20STS%20Token.md).
You should always see the `token request` log.

The application mismatch error can be tested by registering a Service Account config with a scoped MAP STS route that is not one of `chatbot`, `sign_up_service`, `check_in`, or `appointments`.

The missing ICN error can be tested by not including an icn `user_attributes` in your Service Account access_token request.

### Token Request:
`[MAP][SecurityToken][Service] token request, application: chatbot, icn: 1012667122V019349`

### Token Success:
`[MAP][SecurityToken][Service] token success, application: chatbot, icn: 1012667122V019349`

### Application Mismatch:

`[MAP][SecurityToken][Service] token failed, application mismatch detected -- { :application => :foobar, :icn => "1012667122V019349" }`

### missing ICN:

`[MAP][SecurityToken][Service] token failed, ICN not present in access token -- { :application => :foobar }`

### Other MAP STS Client Error:

`[MAP][SecurityToken][Service] token failed, client error, status: , application: chatbot, icn: 1012667122V019349, context: {:error=>nil}`

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature

### CODE REVIEW

File: app/controllers/v0/map_services_controller.rb [hunk 1]
```diff
@@ -7,10 +7,8 @@ class MapServicesController < SignIn::ServiceAccountApplicationController
     service_tag 'identity'
     # POST /v0/map_services/:application/token
     def token
-      unless (icn = @service_account_access_token.user_attributes['icn'])
-        raise MAP::SecurityToken::Errors::MissingICNError
-      end
-
+      icn = @service_account_access_token.user_attributes['icn']
+ \
```
Comment (dickdavis) [inline app/controllers/v0/map_services_controller.rb:11]: ✂️

**Review by dickdavis — COMMENTED** [review]
LGTM, just a small detail to clean-up

File: lib/map/security_token/service.rb [hunk 2]
```diff
@@ -37,7 +47,9 @@ def parse_response(response, application, icn)
           expiration: Time.zone.now + response_body['expires_in']
         }
       rescue => e
-        raise e, "#{config.logging_prefix} token failed, response unknown, application: #{application}, icn: #{icn}"
+        message = "#{config.logging_prefix} token failed, response unknown"
+        Rails.logger.error(message, application:, icn:)
```
[the remark this answers is not in the document]
Comment (bramleyjl) [reply lib/map/security_token/service.rb:51]: I don't believe so, this rescue only hits what happens in this function, so if something goes wrong with the `JSON.parse` or the response body is missing a `access_token` or `expires_in` attribute.

The other error handling/logging is for the 3 specific errors, a `ClientError` if something goes wrong with the POST request, an `ApplicationMismatchError`, and a `MissingICNError`.

[the remark this answers is not in the document]
Comment (bramleyjl) [reply lib/map/security_token/service.rb:51]: Confirmed that this is the only point the MAP service or controller will log an error:
`Rails -- [MAP][SecurityToken][Service] token failed, response unknown -- { :application => :chatbot, :icn => "1012667122V019349" }`

**Review by dickdavis — APPROVED** [review]

**Pull request merged**